orchagent/switchorch.cpp

```diff
@@ -718,6 +951,70 @@ void SwitchOrch::restartCheckReply(const string &op, const string &data, std::ve
     checkRestartReadyDone();
 }
 
+void SwitchOrch::onSwitchAsicSdkHealthEvent(sai_object_id_t switch_id,
+                                            sai_switch_asic_sdk_health_severity_t severity,
+                                            sai_timespec_t timestamp,
+                                            sai_switch_asic_sdk_health_category_t category,
+                                            sai_switch_health_data_t data,
+                                            const sai_u8_list_t &description)
+{
+    std::vector<swss::FieldValueTuple> values;
+    const string &severity_str = switch_asic_sdk_health_event_severity_reverse_map.at(severity);
+    const string &category_str = switch_asic_sdk_health_event_category_reverse_map.at(category);
+    string description_str;
+    const std::time_t &t = timestamp.tv_sec;
+    stringstream time_ss;
+    time_ss << std::put_time(std::localtime(&t), "%Y-%m-%d %H:%M:%S");
+
+    switch (data.data_type)
+    {
+    case SAI_HEALTH_DATA_TYPE_GENERAL:
+    {
+        vector<uint8_t> description_with_terminator(description.list, description.list + description.count);
+        // Add the terminate character
+        description_with_terminator.push_back(0);
+        description_str = string(reinterpret_cast<char*>(description_with_terminator.data()));
+        // Remove unprintable characters but keep CR and NL
+        description_str.erase(std::remove_if(
```

Comment:
why is there unprintable char in a description data? just in case?

Reply:
yes. just in case.

Reply:
should we log an warning or error in such case? SAI should not pass un-printable chars.

Reply:
I do not think it is worth a warning or error. We can put a piece of `info` message here in case there have been some characters removed.

Reply:
notice?

Reply:
ok. sounds good.

Reply:
updated.